src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +18,72 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+        try {
+            M = Integer.parseInt(argument); // conversion string to int
+        } catch (NumberFormatException e) {
+            System.out.println("'" + argument + "'" + " to " + UsefulConstants.WRONG_ARGUMENT + " - to nie liczba");
+            throw new IllegalArgumentException(e);
+        }
+        if (M > UsefulConstants.MAX_UPPER_BOUND || M < 1) {
+            System.out.println(M + " to " + UsefulConstants.WRONG_ARGUMENT + " - jest spoza zakresu <1, " + UsefulConstants.MAX_UPPER_BOUND + ">");
+            throw new IllegalArgumentException();
+        }
     }
-
     public void play() {
-        //TODO: Implement the method that executes the game session
+        System.out.println("Zagrajmy. Zgadnij liczbę z zakresu <1," + M + ">"); // range of numbers
+
+        Random random = new Random();
+        int randomNumber = random.nextInt(M) + 1; // generate random number <1,M>
+        int L = (int) Math.abs((Math.log10(M)/Math.log10(2)) + 1); // calculate formula
+
+
+        boolean guessedNumber = false;
```

Comment:
_clean code_ nazwa _guessedNumber_ sugeruje, że będziemy przechowywali tutaj int'a. Gdy przechowywujemy sobie jakiegoś _booleana_ to przeważnie ta zmienna zaczyna się na _is..._, _has..._, _are..._, itp. tutaj przykładowo mogłoby być _isNumberGuessed_